util/network/src/connection.rs

```diff
@@ -158,8 +158,8 @@ impl<Socket: GenericSocket> GenericConnection<Socket> {
 			}
 			if self.send_queue.is_empty() {
 				self.interest.remove(Ready::writable());
-				io.update_registration(self.token)?;
 			}
+			io.update_registration(self.token)?;
```

Comment:
This fixes a race condition when the sending queue is being processed and another packet is added to the queue. The registration might have not been updated in such case.